is_domain.go

```diff
@@ -31,14 +33,19 @@ func IsDomain(s string) bool {
 	if strings.HasSuffix(s, ".") {
 		s = s[:len(s)-1]
 	}
-
 	split := strings.Split(s, ".")
 	tld := split[len(split)-1]
-
 	if !IsTLD(tld) {
 		return false
 	}
-
 	s = strings.ToLower(s)
+	// allow matching of dnslink domains
+	// the following only removes `_dnslink.``
+	// if it is the leading value.
+	// it will remove from _dnslink.example.com
+	// but will not remove from example._dnslink.com
+	if strings.Contains(s, "_dnslink.") {
+		s = strings.TrimPrefix(s, "_dnslink.")
+	}
```

Comment:
This library is a general-purpose "is domain" library so we should probably do this elsewhere if we actually need it.

Reply:
It makes sense what you're saying, however missing this chunk of code appears to have been what was causing `go-dnslink` to be unable to resolve `_dnslink.` records, as it would pass the TLD validation check, but then fail on the regexp check, presumably due to it not containing an `_`. 

My other inclination was to adjust the regexp to allow for `_` however, that would cause a larger problem I think of records like `foo.b_ar.com` being valid

Reply:
Actually after thinking about this more you are indeed correct, will remove.